src/app/_utils/categoryUtils.ts

```diff
@@ -81,3 +83,21 @@ export function getCategoryLabel({
 
   return option ? option.label : category
 }
+
+export function createCategorySchema(
+  validCategoryOptions: string[],
+): z.ZodTypeAny {
+  if (validCategoryOptions.length === 0) {
+    return z.never()
+  } else if (validCategoryOptions.length === 1) {
+    return z.literal(validCategoryOptions[0])
+  } else {
+    return z.union(
+      validCategoryOptions.map((option) => z.literal(option)) as [
+        z.ZodLiteral<string>,
+        z.ZodLiteral<string>,
+        ...z.ZodLiteral<string>[],
+      ],
+    )
+  }
+}
```

Comment:
@CharlyMartin - I'm going to merge this as is now - But I'd still like you review this. I wasn't able to implement it cleaner.

Reply:
@mirhamasala, after reading about zod emums, I was able to refactor the above code into the following, which reads better, in my opinion.

I tested it with invalid categories and values and it does fail.

```ts
export function createCategorySchema(cmsCategories: Array<string>) {
  if (cmsCategories.length === 0) {
    return z.never()
  } else if (cmsCategories.length === 1) {
    return z.literal(cmsCategories[0])
  } else {
    const [first, second, ...rest] = cmsCategories
    return z.enum([first, second, ...rest])
  }
}
```

What do you think? Should I push this update to the other open zod PR?